core/src/test/java/org/apache/spark/launcher/SparkLauncherSuite.java

```diff
@@ -175,6 +175,33 @@ public void testChildProcLauncher() throws Exception {
       .setConf(SparkLauncher.DRIVER_EXTRA_CLASSPATH, System.getProperty("java.class.path"))
       .addSparkArg(opts.CLASS, "ShouldBeOverriddenBelow")
       .setMainClass(SparkLauncherTestApp.class.getName())
+      .autoShutdown(true)
```

Comment:
I remember mentioning that this isn't testing anything, maybe I forgot to send the comment...

But since the parent process is not dying, this isn't actually doing anything. If you want to test this here (and not just with the YARN tests you added), you'll need to write a different test.